ostd/src/mm/io.rs

```diff
@@ -288,34 +312,32 @@ unsafe fn memcpy_fallible(dst: *mut u8, src: *const u8, len: usize) -> usize {
 /// and physical address level. There is not guarantee for the operation results
 /// of `VmReader` and `VmWriter` in overlapping untyped addresses, and it is
 /// the user's responsibility to handle this situation.
-pub struct VmReader<'a, Space = KernelSpace> {
+pub struct VmReader<'a, Fallibility = Fallible> {
     cursor: *const u8,
     end: *const u8,
-    phantom: PhantomData<(&'a [u8], Space)>,
+    phantom: PhantomData<(&'a [u8], Fallibility)>,
 }
 
 macro_rules! impl_read_fallible {
-    ($read_space:ty, $write_space:ty) => {
-        impl<'a> VmReader<'a, $read_space> {
+    ($reader_fallibility:ty, $writer_fallibility:ty) => {
+        impl<'a> FallibleRead<$writer_fallibility> for VmReader<'a, $reader_fallibility> {
             /// Reads all data into the writer until one of the three conditions is met:
             /// 1. The reader has no remaining data.
             /// 2. The writer has no available space.
             /// 3. The reader/writer encounters some error.
-            ///
```

Comment:
I am wondering if comments are needed or useful here at all. The trait declaration already says the same thing.